src/rgw/rgw_sal.h

```diff
@@ -307,12 +307,19 @@ class RGWObject {
 		 lastmod(nullptr), target_obj(nullptr) {}
       } params;
 
+      struct Result {
+        rgw_raw_obj head_obj;
+
+        Result() : head_obj() {}
+      } result;
+
       explicit ReadOp(RGWObject *_source) : source(_source) {}
       virtual ~ReadOp() = default;
 
       virtual int prepare(optional_yield y) = 0;
       virtual int read(int64_t ofs, int64_t end, bufferlist& bl, optional_yield y) = 0;
       virtual int iterate(int64_t ofs, int64_t end, RGWGetDataCB *cb, optional_yield y) = 0;
+      virtual int get_manifest(RGWObjManifest **pmanifest, optional_yield y) = 0;
```

Comment:
is this generally possible?  what is a non-RGWRados manifest?

Reply:
It's used by GetObjLayout, and sent out over the wire, so it seems that it's part of the non-RADOS format.   All the fields in the RGWObjManifest class seem to be store agnostic, so other stores can implement it.

Reply:
i believe that GetObjLayout is only used for ragweed tests. we could probably have that API return a not-supported error if the backend isn't rados